src/Dto/AbstractStimulusDto.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony WebpackEncoreBundle package.
+ * (c) Fabien Potencier <fabien@symfony.com>
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Dto;
+
+use Twig\Environment;
+
+abstract class AbstractStimulusDto implements \Stringable
+{
+    /**
+     * @var Environment
+     */
+    private $env;
+
+    public function __construct(Environment $env)
+    {
+        $this->env = $env;
+    }
+
+    abstract public function toArray(): array;
+
+    protected function getFormattedControllerName(string $controllerName): string
+    {
+        return $this->escapeAsHtmlAttr($this->normalizeControllerName($controllerName));
+    }
+
+    protected function getFormattedValue($value)
+    {
+        if ($value instanceof \Stringable || (\is_object($value) && \is_callable([$value, '__toString']))) {
+            $value = (string) $value;
+        } elseif (!is_scalar($value)) {
+            $value = json_encode($value);
+        } elseif (\is_bool($value)) {
+            $value = $value ? 'true' : 'false';
+        }
+
+        return $this->escapeAsHtmlAttr($value);
+    }
+
+    /**
+     * Normalize a Stimulus Value API key into its HTML equivalent ("kebab case").
+     * Backport features from symfony/string.
+     *
+     * @see https://stimulus.hotwired.dev/reference/values
+     */
+    protected function normalizeKeyName(string $str): string
```

Comment:
Unless I'm missing something, I think this is only used in one sub-class, so it could be moved there.